### ISSUE TITLE
Similarity embedding Plot lines

### ISSUE THREAD

**Issue by Matthewkm** [materight/RepNet-pytorch#4]
Hi, 

Thanks for the implementation.

I was just wondering if the horizontal and vertical dark lines (like a grid) on the embedding plots are correct?
![example5_tsm](https://github.com/materight/RepNet-pytorch/assets/43727012/942d343f-c6a0-48da-8a44-67a1552a1d8d)

They don't seem to be present on any of the official implementation examples https://sites.google.com/view/repnet

I've had a look through the code but can't really work out why they're there, other than maybe some issue with the frame embeddings or frame sampling? 

Cheers, Matthew.

**Comment by materight** [comment]
Hi, I'm not 100% siure but I think the dark bars are caused by the way the embeddings are extracted. There is a temporal convolution with a kernel of 3x3x3 + 1 padding, so the embeddings of the first and last frame of a 64 frames chunk might be affected by the padding.

I think the examples they show are only taken from a single 64 video chunk, so you don't see the dark bars. But they didn't release the code for it, so it's hard to say.

**Comment by Matthewkm** [comment]
Hi, 

Yes I think you're right with the temporal_conv(x) in the extract features causing the problem. 

I assume the zero padding in the time dimension, along the dilated conv and lack of similarity between frames away from the centre diagonal results in the areas of low values around the borders.

The padding is 3 in the temporal dimension, so I guess a simplest fix for getting longer than 64 frames is to just to remove the first and last 3 values from each embedding before concatenating them for plotting - although not ideal.

Cheers.

**Comment by materight** [comment]
If you don't want to lose those frames in the plot you could also recompute the embeddings including their neighbors to evoid the effect of the padding. So to have a correct embedding for frame 64, you could do a forward pass using frames [..., 63, 64, 65, ...].

**Comment by QQQHHHBBB** [comment]
where to find checkpoints/pytorch_model.pth

**Comment by materight** [comment]
You can run `convert_weights.py` to convert the original tensorflow weights to pytorch.
